executor_manager/services/task_heartbeat_manager.py

```
@@ -412,10 +394,22 @@ async def _handle_task_dead(
             )
             return
 
-        # Step 1: Check container status before deciding action
-        container_status = get_container_status(executor_name)
+        # Step 1: Check container/pod status via ExecutorDispatcher
+        try:
+            executor = ExecutorDispatcher.get_executor(EXECUTOR_DISPATCHER_MODE)
+            container_status = executor.get_container_status(executor_name)
+        except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)